### PR TITLE
feat(ui): Add RAM and Cores info to LXD project tab

### LINKED ISSUE #2230: Add LXD projects resource card

### PR DESCRIPTION
**Pull request by Caleb-Ellis**
## Done

- Created `RamResources` and `CoreResources` components which have mostly been lifted from `PodResourcesCard`, with some changes to work with the new data structure. The plan is to create all the resource sections as individual components then remove `PodResourcesCard`.
- Added `ProjectResourcesCard` to the projects tab, which currently includes the project's RAM and Core data.

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Go to the project tab of a LXD pod.
- Check that the RAM and core data is shown.
- Resize the viewport and check that the layout changes properly.

## Fixes

Fixes #2230

### CODE REVIEW

File: ui/src/app/kvm/views/KVMDetails/RamResources/_index.scss [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        &:nth-child(3) {
+          padding-right: 0;
+          width: 50%;
+        }
+      }
+    }
+
+    // Additional styles for when the component changes layout depending on the
+    // viewport size. Otherwise, the component keeps the "mobile" styling.
+    &.dynamic-layout {
```
Comment (huwshimi) [inline ui/src/app/kvm/views/KVMDetails/RamResources/_index.scss:45]: Same question here about `--dynamic-layout`.

**Review by huwshimi — APPROVED** [review]
Looks good, a couple of small questions.

File: ui/src/app/kvm/views/KVMDetails/CoreResources/_index.scss [hunk 2]
```diff
@@ -0,0 +1,37 @@
+@mixin CoreResources {
+  .core-resources {
+    display: flex;
+    flex-direction: column;
+    padding: $spv-inner--medium $sph-inner;
+
+    // Additional styles for when the component changes layout depending on the
+    // viewport size. Otherwise, the component keeps the "mobile" styling.
+    &.dynamic-layout {
```
[the remark this answers is not in the document]
Comment (Caleb-Ellis) [reply ui/src/app/kvm/views/KVMDetails/CoreResources/_index.scss:9]: Yeah, that *would* be proper BEM. Changing it now

File: ui/src/app/kvm/views/KVMDetails/RamResources/RamResources.tsx [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+    free: number; // B
+    pageSize?: number;
+  };
+};
+
+/**
+ * Returns a string with the formatted byte value and unit, e.g 1024 => "1KiB"
+ *
+ * @param memory - the memory in bytes
+ * @returns {string}
```
[the remark this answers is not in the document]
Comment (Caleb-Ellis) [reply ui/src/app/kvm/views/KVMDetails/RamResources/RamResources.tsx:24]: Got it!

**Pull request merged**